### PR TITLE
xray & SSM environment variables

### PR DESCRIPTION
**Pull request by schmod**
This PR provides the infrastructure necessary to run the AWS X-Ray daemon on a standalone EC2 instance.  (Routing unicast UDP traffic to a ECS container is difficult, so the standalone instance ended up being the best/easiest option)

Go check out [the PR](https://github.com/FoundryAI/hud-ai-users-api/pull/46) in users-api to see how to instrument an API service with xray, and check out the [dashboard](https://console.aws.amazon.com/xray/home?region=us-east-1).  

---

This PR *also* includes the infrastructure necessary to pass custom/arbitrary environment variables to containers using the [EC2 Systems Manager Parameter Store](http://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-paramstore.html).  This service largely fulfills the role that we had envisioned for Consul/Vault, but should be considerably simpler to set up and use.  

These variables can be specified in Terraform, or manually created in the AWS console.  To manually create a new environment variable in [the console](https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#Parameters:sort=Name), simply add a new SSM Parameter with the name: `/staging/users-api-staging/env/VAR_NAME`, and `VAR_NAME` will automagically be loaded into your environment if you include the following snippet at the top of your app:

```js
require('aws-param-env').load(`/${process.env.ENVIRONMENT}/users-api-${process.env.ENVIRONMENT}/env`);
```

In all likelihood, we'll want to manually manage the `HUD_AI_CLIENT_ID` and `HUD_AI_CLIENT_SECRET` variables, given that there's already some manual setup involved with those (as they need corresponding entries in the auth db).

### CODE REVIEW

File: api-service/service/main.tf [hunk 1]
```diff
@@ -109,63 +109,50 @@ resource "aws_ecs_task_definition" "main" {
...
+    PORT = "${var.port}"
+    ENVIRONMENT = "${var.environment}"
+    RDS_DB_NAME = "${var.rds_db_name}"
+    RDS_HOSTNAME = "${var.rds_hostname}"
+    RDS_USERNAME = "${var.rds_username}"
+    DYNAMODB_ENDPOINT = "dynamodb.us-east-1.amazonaws.com"
+    SNS_ENDPOINT = "sns.us-east-1.amazonaws.com"
+    AWS_ACCOUNT_ID = "${data.aws_caller_identity.current.account_id}"
+    AWS_ACCESS_KEY = "${var.aws_access_key}"
+    HUD_AI_BASE_API_URL = "https://${var.environment}.api.hud.ai/v1"
```
Comment (njgerner) [inline api-service/service/main.tf]: let's remove this, the idea here is we can use the infrastructure repo for other projects (keep everything HUD agnostic)

File: xray/main.tf [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  key_name = "${var.environment}"
+  subnet_id = "${var.subnet_id}"
+  vpc_security_group_ids = [
+    "${aws_security_group.xray.id}"
+  ]
+  monitoring = true
+  iam_instance_profile = "${aws_iam_instance_profile.xray.id}"
+  user_data = <<USERDATA
+#!/bin/bash
+curl https://s3.dualstack.us-east-2.amazonaws.com/aws-xray-assets.us-east-2/xray-daemon/aws-xray-daemon-2.x.rpm -o /home/ec2-user/xray.rpm
```
Comment (njgerner) [inline xray/main.tf]: us-east-2?

File: api-service/service/main.tf (hunk 1 above)
Comment (schmod) [reply api-service/service/main.tf]: sounds good

File: xray/main.tf (hunk 2 above)
Comment (schmod) [reply xray/main.tf]: ¯\_(ツ)_/¯ that's what the docs said.

**Pull request merged**